src/context.rs

```diff
@@ -39,6 +39,8 @@ pub struct Context {
     pub bob: Arc<RwLock<BobStatus>>,
     pub last_smeared_timestamp: Arc<RwLock<i64>>,
     pub running_state: Arc<RwLock<RunningState>>,
+    /// Mutex to avoid generating the key for the user more than once.
+    pub generating_key_mutex: Mutex<()>,
```

Comment:
it might be more elegant and efficient to use an AtomicBool that gets set to true when key was loaded or generated successfully the first time

Reply:
For that I'd have to manually implement a spinlock though as I no longer want the function to return early while another thread is generating a key.  I think mutex is nicely suited and this isn't a performance critical path.